From f2d8b6bf2eee4023736307f4267c976acd99829b Mon Sep 17 00:00:00 2001
From: Tzu-Mainn Chen <tzumainn@redhat.com>
Date: Thu, 1 May 2014 15:42:03 +0200
Subject: [PATCH] Default GlanceLogFile template parameter value

The default for GlanceLogFile should be '', not an empty string;
however the UI code reads the default as an empty string.

Other parameters that default '' *should* be an empty string, so
there's no universal workaround on the UI side.

Change-Id: I4ba2fa76205f9d61572f7376f7dbaabde939a452
---
 .../infrastructure/overcloud/workflows/undeployed_configuration.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tuskar_ui/infrastructure/overcloud/workflows/undeployed_configuration.py b/tuskar_ui/infrastructure/overcloud/workflows/undeployed_configuration.py
index 2ec1ed4..aa7ab1c 100644
--- a/tuskar_ui/infrastructure/overcloud/workflows/undeployed_configuration.py
+++ b/tuskar_ui/infrastructure/overcloud/workflows/undeployed_configuration.py
@@ -60,6 +60,13 @@ class Action(horizon.workflows.Action):
         for name, data in parameters:
             self.fields[name] = make_field(name, **data)
 
+    def clean(self):
+        # this is a workaround for a single parameter
+        if 'GlanceLogFile' in self.cleaned_data:
+            if not self.cleaned_data['GlanceLogFile']:
+                self.cleaned_data['GlanceLogFile'] = u'\'\''
+        return self.cleaned_data
+
 
 class Step(horizon.workflows.Step):
     action_class = Action
-- 
1.8.1.4

